tests/unit/introspection.tcl

```diff
@@ -299,6 +299,20 @@ start_server {tags {"introspection"}} {
         set _ $res
     } {*"set" "foo"*"get" "foo"*}
 
+    test {MONITOR should support RESP3 protocol} {
+        set rd [valkey_deferring_client]
+        $rd HELLO 3
+        $rd read ; # Consume the HELLO reply
+
+        $rd monitor
+        $rd read ; # Consume the MONITOR reply
+
+        r set foo bar
+
+        assert_match {monitor*"set"*"foo"*"bar"*} [$rd read]
```

Comment:
To check that we actually get RESP3 pushes, we could use `$rd readraw 1` and verify that we get the correct RESP3 encoded replies.

We should probably also check that we can send regular commands (for example PING) while monitoring.

Reply:
> We should probably also check that we can send regular commands (for example PING) while monitoring.
that's was good idea, I think I found a bug (I am playing with my toy client https://github.com/KowalczykBartek/respy), let me fix your comment, write some test and describe behaviour with monitor+regular command.

btw, thanks for super fast reply, in redis repo it took 2years ;)

Reply:
The answer to the philosophical question is just boring, not very philosophical. Monitor clients are flagged as replicas internally. I don't know why it's like that, but that's why you can't use GET. My guess is that in some old implementation it was the same code. We can change it if nothing breaks.

Can you try to find out by reading source code? If we don't flag monitor clients as replicas, do any test cases break?

Reply:
```
Can you try to find out by reading source code? 
```
I will play around this and get back :)

Reply:
@zuiderkwast do you know maybe, if I can run some subset of tests, single unit tests file against my standalone redis ?

Reply:
Yes it's possible.

`./runtest --help` is useful.

`--single unit/introspection` to run one test suite.

Reply:
The test suite normally starts the server. If you want to run against an already started server, I think it's possible, but I don't know exactly how.

Reply:
```
./runtest --port 6379 --host localhost --single unit/introspection
./src/valkey-server --enable-debug-command yes  
```
gave me what I wanted :)

Reply:
- I fixed formatting 
- used a readraw in existing test and added new for monitor+tracking combination

I also played around
```c
c->flag.replica = 0
```
but this resulted in failing tests because of 
```
------ STACK TRACE ------
EIP:
0   valkey-server                       0x000000010458ce40 clientHasPendingReplies + 200

Backtrace:
0   libsystem_platform.dylib            0x000000018ec60184 _sigtramp + 56
1   valkey-server                       0x000000010458ce2c clientHasPendingReplies + 180
2   valkey-server                       0x000000010458c51c prepareClientToWrite + 276
3   valkey-server                       0x000000010458ac14 addReply + 44
4   valkey-server                       0x00000001045a471c replicationFeedMonitors + 932
5   valkey-server                       0x000000010456b230 call + 1928
6   valkey-server                       0x000000010456cc20 processCommand + 3768
7   valkey-server                       0x0000000104585438 processInputBuffer + 516
8   valkey-server                       0x000000010458486c readQueryFromClient + 92
9   valkey-server                       0x000000010469c3a8 connSocketEventHandler + 180
10  valkey-server                       0x0000000104551ac0 aeProcessEvents + 372
11  valkey-server                       0x0000000104579fbc main + 19392
12  dyld                                0x000000018e8a8274 start + 2840
```
And I was unable to figure out what is wrong ... I used to Java's stacktraces :/ But I tried this approach https://github.com/valkey-io/valkey/pull/1426/commits/2c6304cee7693b1d6ad75a66cb6c8eb3bae108c9#diff-1abc5651133d108c0c420d9411925373c711133e7748d9e4f4c97d5fb543fdd9R4265 and it kinda works :D Unfortunately I am unable to say its correct or no

Reply:
I think you can use `addr2line` to get the line numbers.